### PR TITLE
Apps reorder

### PR DESCRIPTION
**Pull request by zxan1285**
## ✏️ Changes
  Added new feature, which allow customers to rearrange apps order, as been requested in support ticket (see references).
Also includes #43 and few more fixes for dev environment.
  
## 🔗 References
  SF: https://auth0.lightning.force.com/lightning/r/Case/5001G00000btLKoQAM/view
  
## 📷 Screenshots
screenshot

## 🎯 Testing
✅ This change has been tested in a Webtask
🚫 This change has unit test coverage
🚫 This change has integration test coverage
🚫 This change has been tested for performance
  
## 🚀 Deployment
✅ This can be deployed any time

### CODE REVIEW

File: server/lib/applications.js [hunk 1]
```diff
@@ -40,6 +40,34 @@ export const saveApplication = (id, body, storage) => new Promise((resolve, reje
...
+        ids.forEach((key) => {
+          reordered[key] = data.applications[key];
+        });
+
+        data.applications = reordered; // eslint-disable-line no-param-reassign
+
+        return storage.write(data);
+      }
+
+      return null;
```
Comment (fyockm) [inline server/lib/applications.js]: should this throw an error of some sort?

File: server/lib/applications.js [hunk 2]
```diff
@@ -40,6 +40,34 @@ export const saveApplication = (id, body, storage) => new Promise((resolve, reje
     .catch(reject);
 });
 
+/*
+ * Move application to rearrange order.
+ */
+export const moveApplication = (id, direction, storage) =>
```
Comment (fyockm) [inline server/lib/applications.js]: need to add some tests for this function

File: client/actions/application.js [hunk 3]
```diff
@@ -160,6 +160,27 @@ export function cancelDeleteApplication() {
   };
 }
 
+export function moveApplication(appId, moveUp, onSuccess) {
```
Comment (fyockm) [inline client/actions/application.js]: why pass `moveUp` as a boolean? would it make more sense to just have this as the string `up` or `down`?

related https://github.com/auth0-extensions/auth0-sso-dashboard-extension/pull/44/files#diff-a8584ecb771db6e02bdc2fea89fce1d1R51

File: client/components/Applications/ApplicationsTable.jsx [hunk 4]
```diff
@@ -43,8 +40,28 @@ export default class ApplicationsTable extends Component {
...
 
+  renderMoveButton(id, direction, row, total) {
+    const icon = direction === 'up' ? 'icon-budicon-462' : 'icon-budicon-460';
+    const enabled = (direction === 'up' && row > 1) || (direction === 'down' && row < total);
+    return (
+      <li title={direction} data-toggle="tooltip">
+        <a
+          href="#" onClick={(e) => {
+            if (enabled) {
+              this.props.moveApplication(id, direction === 'up');
```
Comment (fyockm) [inline client/components/Applications/ApplicationsTable.jsx]: why convert direction to bool?

File: server/lib/applications.js (hunk 1 above)
Comment (zxan1285) [reply server/lib/applications.js]: It means that we're trying to move first element to the very beginning, or last element to the very end of array. No error, but also no actions required.

File: server/lib/applications.js (hunk 2 above)
Comment (zxan1285) [reply server/lib/applications.js]: Done.

File: client/components/Applications/ApplicationsTable.jsx (hunk 4 above)
Comment (zxan1285) [reply client/components/Applications/ApplicationsTable.jsx]: Fixed.

File: client/actions/application.js (hunk 3 above)
Comment (zxan1285) [reply client/actions/application.js]: Fixed.

File: tests/server/lib/applications.tests.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+const storage = {
+  read: () => Promise.resolve(storageData),
+  write: (data) => {
+    storageData = data;
+    Promise.resolve();
+  }
+};
+
+describe('applications', () => {
```
Comment (fyockm) [inline tests/server/lib/applications.tests.js]: what if the `id` is not found in the list of applications?
what if the `direction` is not `up` or `down`?

**Pull request merged**